tools/ea/Ea/Genesis.hs

```diff
@@ -49,87 +53,51 @@ module Ea.Genesis
 import Control.Lens
 
 import Data.Text
+import Data.Word
 
 import Chainweb.Graph
 import Chainweb.Test.TestVersions
 import Chainweb.Version
 import Chainweb.Version.Development
+import Chainweb.Version.FastDevelopment
 import Chainweb.Version.Mainnet
 import Chainweb.Version.Testnet
 
 
 -- ---------------------------------------------------------------------- --
 -- Genesis Tx Data
 
--- | Type-safe representation of chain ids
--- for a 10-chain graph.
---
-data GChainId
-    = Zero
-    | One
-    | Two
-    | Three
-    | Four
-    | Five
-    | Six
-    | Seven
-    | Eight
-    | Nine
-    | N
-    | KAD
-    deriving (Eq, Ord, Enum)
-
-instance Show GChainId where
-  show = \case
-    Zero -> "0"
-    One -> "1"
-    Two -> "2"
-    Three -> "3"
-    Four -> "4"
-    Five -> "5"
-    Six -> "6"
-    Seven -> "7"
-    Eight -> "8"
-    Nine -> "9"
-    N -> "N"
-    KAD -> "KAD"
+-- A range of chain IDs [l,r]
+data ChainIdRange
+    = ChainIdRange !Word32 !Word32
+    deriving (Eq, Ord)
+
```

Comment:
should there be a smart constructor and some other utilities?:
```haskell
mkChainIdRange :: Word32 -> Word32 -> ChainIdRange
mkChainIdRange l r = ChainIdRange (min l r) (max l r)

singleton :: Word32 -> ChainIdRange
singleton x = ChainIdRange x x
```

Also, there are a lot of "magic numbers" below. They could instead be repeatably used constants.

Reply:
I'd take a `mkChainIdRange` that threw an error on an invalid range. I'll take a look at those constants.